aioshelly/rpc_device.py

```diff
@@ -87,7 +87,6 @@ def _on_notification(
         if not self._initializing and not self.initialized:
             loop = asyncio.get_running_loop()
             loop.create_task(self._async_init())
-            return
```

Comment:
The packet that trigger the async init is a `status` packet, before this change we dropped this packet and fetch it again here, so I added a flag not to fetch it twice:
https://github.com/home-assistant-libs/aioshelly/blob/9ea9869d09c77b71181cce30d6be45be7ea7b6cc/aioshelly/rpc_device.py#L134-L135

Since these are battery operated devices, power consumption is important so I want to minimize the communication with the device, in addition device goes back to sleep shortly after sending the status so we need to finish the communication with the device as fast as possible.
When we introduced the support for re-auth flow in https://github.com/home-assistant/core/pull/78786 we also found out that with all of the sleeping devices except for Shelly TRV (climate device) we only need the status packet in order to update the device data and make it operational (at startup we restore the sensors).

This means that with this single packet that is pushed to us by the device we can already make the device working, even if the device authorization settings changed and fetching `config` (Gen2) or `settings` (Gen1) fails, this is why we silently log the auth error but let the device be marked as initialized. It was not planned but we found it during testing. Dropping this packet means we must successfully fetch it from the device.

This change has a big impact of the user experience, even if auth data in the device changed it does not need to be updated in Home Assistant unless we need it (and in this case it will trigger re-auth flow), so devices will continue to work, updating auth data for a battery operated device in Home Assistant is not user friendly, after starting the re-auth flow the user has to manually wake up the device using a button and the new details, which we can avoid.

Reply:
Thanks for the explanation.